tree-diff

```diff
@@ -77,6 +85,13 @@ def diff_aux(dir_fd1, dir_fd2, path, report):
                 stat2 = os.stat(dirent.name, dir_fd=dir_fd2, follow_symlinks=False)
             except FileNotFoundError:
                 report["deleted_files"] += [os.path.join(path, dirent.name)]
+                if dirent.is_dir(follow_symlinks=False):
+                    try:
+                        child_fd = os.open(dirent.name, os.O_DIRECTORY, dir_fd=dir_fd1)
+                    except OSError:
+                        continue
```

Comment:
Which `OSError` do you want to ignore? I feel like this except handler might be too broad. Maybe something like [this](https://github.com/osbuild/osbuild/blob/master/osbuild/objectstore.py#L16)?